app/cni/pkg/cni/main.go

```diff
@@ -0,0 +1,256 @@
+	var result *current.Result
+	if conf.PrevResult == nil {
+		result = &current.Result{
+			CNIVersion: current.ImplementedSpecVersion,
+		}
+	} else {
+		// Pass through the result for the next plugin
+		result = conf.PrevResult
+	}
+	logger.Info("result", "result", result)
+	return types.PrintResult(result, conf.CNIVersion)
+}
+
+func excludeByMissingSidecarInjectedAnnotation(annotations map[string]string) bool {
+	excludePod := false
+	val, ok := annotations[metadata.KumaSidecarInjectedAnnotation]
+	if !ok || val != "true" {
+		excludePod = true
+	}
+	return excludePod
+}
+
+func isInitContainerPresent(initContainersMap map[string]struct{}) bool {
+	excludePod := false
+	// Check if kuma-init container is present; in that case exclude pod
+	if _, present := initContainersMap[util.KumaInitContainerName]; present {
+		excludePod = true
+	}
+	return excludePod
+}
+
+func getPodInfoWithRetries(ctx context.Context, conf *PluginConf, k8sArgs K8sArgs) (int, map[string]struct{}, map[string]string, error) {
+	client, err := newKubeClient(*conf)
+	if err != nil {
+		return 0, nil, nil, errors.Wrap(err, "could not create kube client")
+	}
+
+	var containerCount int
+	var initContainersMap map[string]struct{}
+	var annotations map[string]string
+	var k8sErr error
+
+	backoff := retry.WithMaxRetries(podRetrievalMaxRetries, retry.NewConstant(podRetrievalInterval))
+	err = retry.Do(ctx, backoff, func(ctx context.Context) error {
+		containerCount, initContainersMap, annotations, k8sErr = getKubePodInfo(ctx, client, string(k8sArgs.K8S_POD_NAME), string(k8sArgs.K8S_POD_NAMESPACE))
+		if k8sErr != nil {
+			log.Error(k8sErr, "error getting pod info", "retries", podRetrievalMaxRetries)
+			return retry.RetryableError(k8sErr)
+		}
+		log.V(1).Info("pod container info",
+			"count count", containerCount,
+			"initContainers", initContainersMap,
+			"annotations", annotations,
+		)
+		return nil
+	})
+	if err != nil {
+		return 0, nil, nil, errors.Wrap(err, "failed to get pod data")
+	}
+
+	return containerCount, initContainersMap, annotations, nil
+}
+
+func shouldExcludePod(excludedNamespaces []string, podNamespace types.UnmarshallableString) bool {
+	excludePod := false
+	for _, excludeNs := range excludedNamespaces {
+		if string(podNamespace) == excludeNs {
+			excludePod = true
+			break
+		}
+	}
+	return excludePod
+}
+
+func logPrevResult(conf *PluginConf) {
+	var loggedPrevResult interface{}
+	if conf.PrevResult == nil {
+		loggedPrevResult = "none"
+	} else {
+		loggedPrevResult = conf.PrevResult
+	}
+
+	log.V(1).Info("cmdAdd config parsed", "version", conf.CNIVersion, "prevResult", loggedPrevResult)
+}
+
+func cmdCheck(args *skel.CmdArgs) error {
+	return nil
+}
+
+// cmdDel is called for DELETE requests
+func cmdDel(args *skel.CmdArgs) error {
+	return nil
+}
+
+func main() {
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, fmt.Sprintf("kuma-cni %v", kuma_version.Build.Version))
+}
+
+func Run() {
+	main()
```

Comment:
placing main in package that is not `main` is a bit awkward. I'd inline the main here and call this file `run.go`

Reply:
inlined